apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/assessmentQuestion.ts

```diff
@@ -136,6 +143,179 @@ router.post(
       } else {
         res.send({});
       }
+    } else if (req.body.__action === 'bot_grade_assessment') {
```

Comment:
I'd strongly encourage you to move the logic in this code to a file like `apps/prairielearn/src/lib/bot-grading.ts` and call a function from here. This would make the code cleaner, and would make it easier to later on reuse this in other interfaces.